src/mod_auth_pubtkt.c

```diff
@@ -439,6 +444,105 @@ static int cookie_match(void *result, const char *key, const char *cookie) {
 	return 1;
 }
 
+/* Look for a ticket in a header - maybe a cookie, maybe not */
+static char *get_header_ticket(request_rec *r) {
+	auth_pubtkt_dir_conf *conf = ap_get_module_config(r->per_dir_config, &auth_pubtkt_module);
+
+	char *ticket_decoded, *found_ticket;
+	const char *header_name, *header_list_str, *header_value;
+	size_t ticket_len;
+
+	header_list_str = (conf->auth_header_name) ? conf->auth_header_name : MOD_AUTH_PUBTKT_HEADER_NAME;
+
+	if( !*header_list_str ) {
+		ap_log_rerror(APLOG_MARK, APLOG_ERR, APR_SUCCESS, r,
+			"TKTAuthHeader directive is empty"
+		);
+		return NULL;
+	}
+
+	if (conf->debug >= 1) {
+		ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+			"TKT get_header_ticket: looking for ticket in '%s' header(s)",
+			header_list_str
+		);
+	}
+
+	while( *header_list_str && (header_name = ap_getword_conf(r->pool, &header_list_str)) ) {
+
+		if(!header_name) {
+			break;
+		}
+
+		if (conf->debug >= 1) {
+			ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+				"TKT get_header_ticket: inspecting header '%s'",
+				header_name
+			);
+		}
+
+		if( strcasecmp(header_name,"Cookie") == 0 ) {
+			found_ticket = get_cookie_ticket(r);
+		}
+		else {
+			header_value = apr_table_get(r->headers_in,header_name);
+			if(!header_value) {
+				if (conf->debug >= 1) {
+					ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+						"TKT get_header_ticket: auth header missing"
+					);
+				}
+				continue;
+			}
+			ticket_decoded = apr_pstrdup(r->pool,header_value);
+			ticket_len     = strlen(ticket_decoded);
+			if( ticket_len <= 0 ) {
+				if (conf->debug >= 1) {
+					ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+						"TKT get_header_ticket: auth header present but empty"
+					);
+				}
+				continue;
+			}
+
+			int i;
+			/* Replace '+' by ' ' (not handled by ap_unescape_url_keep2f) */
+			for (i = 0; ticket_decoded[i]; i++) {
+				if (ticket_decoded[i] == '+')
+					ticket_decoded[i] = ' ';
+			}
+
+			/* URL-unescape cookie */
+#ifdef APACHE24
+			if (ap_unescape_url_keep2f(ticket_decoded, 1) != 0) {
+#else
+			if (ap_unescape_url_keep2f(ticket_decoded) != 0) {
+#endif
+				ap_log_rerror(APLOG_MARK, APLOG_WARNING, APR_SUCCESS, r,
+					"TKT get_header_ticket: error while URL-unescaping cookie");
+				continue;
+			}
+			if (conf->debug >= 1) {
+				ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+					"TKT get_header_ticket: found '%s'", ticket_decoded);
+			}
+			found_ticket = apr_pstrdup(r->pool, ticket_decoded);
+		}
+
+	    if( found_ticket != NULL ) {
+			if (conf->debug >= 1) {
+				ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+					"TKT get_header_ticket: found ticket in header '%s'", header_name);
+			}
+			return found_ticket;
```

Comment:
If the cookie header appears first, and and has a cookie that matches TKTAuthCookieName, it will be used regardless of having specified TKTAuthHeader from what I can tell.